### DISCUSSION TITLE
Let's find another name for `fc.set`

### DISCUSSION THREAD

**Discussion by dubzzz** [dubzzz/fast-check#1985] [Announcements]
The built-in arbitrary `fc.set` is able to generate arrays of unique values. Unicity is defaulted by `===` (might change for `Object.is`) and can be overriden easily by anything else.

**But** it has one naming issue: it does not produce `Set` but `Array`.

The aim of this discussion is to find a better name for those "arrays of unique values". I was thinking about:
- `fc.uniqueArray`
- `fc.unique`
- _your ideas are more than welcomed_

Renaming it would make future versions of fast-check able to re-use it for real instances of `Set`.

**Comment by jlink** [comment]
IMO uniqueness is an orthogonal aspect of all kinds of collections. So I’d go with some kind of fluent API, e.g. fc.array().withUniqueElements(). But I’m biased, jqwik is doing it like that: https://jqwik.net/docs/current/user-guide.html#uniqueness-constraints

**Reply by dubzzz** [reply]
Indeed a fluent API might be a really cool solution 🤔 

For the moment, it has not been the choice made by fast-check but actually it was mostly to be closer to jsverify. I will definitely have to double how you did on your side and probably also check hypothesis plus others..

**Comment by dubzzz** [comment]
So far, in an attempt to stay aligned with the other existing APIs, I'll probably go for `uniqueArray` with PR https://github.com/dubzzz/fast-check/pull/2616.

But fluent APIs or a `array` able to offer a unique feature will possibly come in the future. So far, I found it pretty handy to be able to build array of unique values with something like `uniqueArray(arb)` instead of `array(arb, {unique: true})` or `array(arb).uniqueValues()` (less fast-check idiomatic than the other one).

**Comment by dubzzz** [comment]
The final choice went for `uniqueArray`. I'll probably re-use `set` for instances of `Set`.
